blockDeleted

captain695 opened the pull request:
@agardella @cburk

Commit:
Added blockDeleted function (without watcher for delete queue)
Added test to make sure block is deleted
Added delete folder constant in client commons

Commit:
Fixed typo in logging in blockDeleted in dn client

Commit:
Added teardown for zk_dn_client_test.cc

Commit:
Merge branch 'develop' into zk-dn-client

Commit:
Added blockDeleted function and test for it

pelmers commented on zookeeper/source/zk_client_common.cc:
```diff
@@ -25,6 +25,7 @@ namespace zkclient {
     const std::string ZkClientCommon::CLASS_NAME = ": **ZkNnCommon** : ";
 	const std::string ZkClientCommon::BLOCK_LOCATIONS = "/block_locations/";
     const std::string ZkClientCommon::BLOCKS = "/blocks";
+    const std::string ZkClientCommon::DELETE_BACKSLASH = "/delete/";
```
🙃

captain695 replied:
Oops yeah I'll delete that

pelmers replied:
oh that was not any review, just noting for amusement that we say backslash for /

pelmers replied:
although now that you mention it i think it's right to delete it since it's never used

Vegetable26 commented on test/zookeeper/zk_dn_client_test.cc:
```diff
@@ -34,17 +37,6 @@ class ZKDNClientTest : public ::testing::Test {
 	ZkClientDn *client;
 };
 
-TEST_F(ZKDNClientTest, RegisterMakesWorkQueues){
```
Just curious, why are we removing this test?

captain695 replied:
I accidentally left it out when merging the two commits. Will add back in

Vegetable26 commented on zookeeper/include/zk_dn_client.h:
```diff
@@ -71,6 +71,14 @@ class ZkClientDn : public ZkClientCommon {
+    * @param size_bytes The number of bytes in the block
+    * @return True on success, false on error.
+    */
+    bool blockDeleted(uint64_t uuid, uint64_t size_bytes);
```
I don't think size_bytes is ever used

captain695 replied:
You are correct. Removed

Vegetable26 commented on zookeeper/source/zk_dn_client.cc:
```diff
@@ -120,6 +119,38 @@ namespace zkclient{
+		bool deleted_correctly = true;
+
+		// Delete block locations
+		if (zk->exists(BLOCK_LOCATIONS + std::to_string(uuid), exists, error_code)) {
```
Can we fold these two deletes into a multi-op?

Vegetable26 replied:
In the future, we will also want to delete the corresponding work_queue/delete/<datanode_id>/<block_id> znode in this multi-op

Also do we have to worry about deleting from wait_for_acks? I'm not super sure, what do you guys think?

captain695 replied:
I asked @nichhk and @cburk and they thought no, but if anyone has a reason for thinking we should chime in

Vegetable26 replied:
Oh kk, sounds good :) I think multi-ops might still be a good idea though

Commit:
Got rid of unnecessary DELETE_BACKSLASH macro
Re-added RegisterMakesWorkQueues test
Removed size_bytes input for blockDeleted function
Put deletes in blockDeleted into a multi-op

captain695 commented:
@nichhk does this all look good?

nichhk commented on zookeeper/source/zk_dn_client.cc:
```diff
@@ -120,6 +119,39 @@ namespace zkclient{
+		auto ops = std::vector<std::shared_ptr<ZooOp>>();
+
+		// Delete block locations
+		if (zk->exists(BLOCK_LOCATIONS + std::to_string(uuid), exists, error_code)) {
```
The exists path and delete path don't match.

nichhk commented on zookeeper/source/zk_dn_client.cc:
```diff
@@ -120,6 +119,39 @@ namespace zkclient{
+		}
+
+		// Delete blocks
+		if (zk->exists(HEALTH_BACKSLASH + id + BLOCKS, exists, error_code)) {
```
The exists path and delete path don't match.

efrederic commented on test/zookeeper/zk_dn_client_test.cc:
```diff
@@ -35,16 +38,17 @@ class ZKDNClientTest : public ::testing::Test {
+    uint64_t size;
+ 
+    std::vector<std::uint8_t> data(sizeof(BlockZNode));
+    std::string path = "/work_queues/replicate/" + dn_id;
```
spacing

Commit:
Correct paths in blockDeleted
Fix spacing

Commit:
Now deletes /block_deleted/<uuid> if it has no children
Added a test for this
